<launch>
    <!-- Camera RBG Image Raw topic -->
    <arg name ="input_image_topic"      default="/camera/rgb/image_raw"/>

    <!-- Select the camera base frame -->
    <arg name ="base_frame_name"        default="camera_link"/>
    <!-- Select the pose_2d topic name -->
    <arg name ="pose_2d_topic_name"     default="/human_pose_2d/pose_array"/>
    <!-- Select the cloud topic name -->
    <arg name ="cloud_topic_name"       default="/camera/depth/points"/>
    <!-- Select the camera_info topic name -->
    <arg name ="camera_info_topic_name" default="/camera/rgb/camera_info"/>

    <!-- Show 2D Pose result image (true) -->
    <arg name ="pose_2d_img_show"       default="false"/>
    <!-- Publish 2D Pose result image (true) -->
    <arg name ="pose_2d_img_pub"        default="true"/>
    <!-- Show 2D Pose result as log in terminal (true) -->
    <arg name ="pose_2d_log_show"       default="false"/>
    
    <!-- Enable 3D Pose detection (true) -->
    <arg name ="pose_3d_detect"         default="true"/>
    <!-- Publish 3D Pose result as topic (true) -->
    <arg name ="pose_3d_topic_pub"      default="false"/>
    <!-- Broadcast 3D Pose result as TF (true) -->
    <arg name ="pose_3d_tf_pub"         default="true"/>
    <!-- Show 3D Pose result as log in terminal (true) -->
    <arg name ="pose_3d_log_show"       default="false"/>

    <arg name ="checkpoint_path"        default="$(find lightweight_human_pose_estimation)/script/checkpoints/checkpoint_iter_370000.pth"/>
    <arg name ="height_size"            default="256"/>
    <arg name ="cpu"                    default="true"/>
    <arg name ="track"                  default="1"/>
    <arg name ="smooth"                 default="1"/>

    <node name="human_pose_2d" pkg = "lightweight_human_pose_estimation" type="human_pose_2d.py" output="screen">
        <param name="checkpoint_path"       type="str"      value="$(arg checkpoint_path)"/>
        <param name="height_size"           type="int"      value="$(arg height_size)"/>
        <param name="cpu"                   type="bool"     value="$(arg cpu)"/>
        <param name="track"                 type="int"      value="$(arg track)"/>
        <param name="smooth"                type="int"      value="$(arg smooth)"/>

        <param name="input_image_topic"     type="str"      value="$(arg input_image_topic)"/>

        <param name="pose_2d_img_show"      type="bool"     value="$(arg pose_2d_img_show)"/>
        <param name="pose_2d_img_pub"       type="bool"     value="$(arg pose_2d_img_pub)"/>
        <param name="pose_2d_log_show"      type="bool"     value="$(arg pose_2d_log_show)"/>
    </node>

    <node name="human_pose_3d" pkg="lightweight_human_pose_estimation" type="human_pose_3d" respawn="false" output="screen">
        <param name="base_frame_name"    value="$(arg base_frame_name)"/>
        <param name="pose_2d_topic_name" value="$(arg pose_2d_topic_name)"/>
        <param name="cloud_topic_name"   value="$(arg cloud_topic_name)"/>
        <param name="camera_info_topic_name" value="$(arg camera_info_topic_name)"/>

        <param name="pose_3d_detect"    type="bool" value="$(arg pose_3d_detect)"/>
        <param name="pose_3d_topic_pub" type="bool" value="$(arg pose_3d_topic_pub)"/>
        <param name="pose_3d_tf_pub"    type="bool" value="$(arg pose_3d_tf_pub)"/>
        <param name="pose_3d_log_show"  type="bool" value="$(arg pose_3d_log_show)"/>
    </node>
</launch>